localstack/services/apigateway/invocations.py

```diff
@@ -192,18 +203,22 @@ def is_api_key_valid(invocation_context: ApiInvocationContext) -> bool:
     ).apigateway
     rest_api = client.get_rest_api(restApiId=invocation_context.api_id)
 
-    if rest_api.get("apiKeySource") != "HEADER":
-        # When the apiKeySource is set to AUTHORIZER, the authorizer is supposed to return the API key as a field
-        # `usageIdentifierKey`
-        # see https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-output.html
-        # Authorizers are only mocked, so we can't validate the key. Return True in that case
-        return True
-
-    api_key = invocation_context.headers.get("X-API-Key")
-    if not api_key:
-        return False
-
-    return validate_api_key(api_key, invocation_context)
+    # The source of the API key for metering requests according to a usage plan.
+    # Valid values are:
+    # - HEADER to read the API key from the X-API-Key header of a request.
+    # - AUTHORIZER to read the API key from the UsageIdentifierKey from a custom authorizer.
+
+    api_key_source = rest_api.get("apiKeySource")
+    match api_key_source:
+        case "HEADER":
+            api_key = invocation_context.headers.get("X-API-Key")
+            return validate_api_key(api_key, invocation_context) if api_key else False
+        case "AUTHORIZER":
+            # When the apiKeySource is set to AUTHORIZER, the authorizer is supposed to return the API key as a field
+            # `usageIdentifierKey`
+            # see https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-output.html
+            # Authorizers are only mocked, so we can't validate the key. Return True in that case
+            return run_authorizer(invocation_context, rest_api.get("authorizer"))
```

Comment:
I think `run_authorizer` does not return `True` when not overridden, which means it will always fail for community

Reply:
yep, after the discussion, I'm happy to not hook it here.